internal/server/cookie.go

```diff
@@ -61,21 +61,21 @@ func deleteCookie(c *gin.Context, name, domain string) {
 }
 
 // exchangeSignupCookieForSession sets the auth cookie on the current host making the request
-func exchangeSignupCookieForSession(c *gin.Context, baseDomain string) {
-	key := getRequestContext(c).Authenticated.AccessKey // this should have been set by the middleware
-	if key != nil {
-		signupCookie, err := getCookie(c.Request, cookieSignupName)
-		if err != nil {
-			logging.L.Trace().Err(err).Msg("failed to find signup cookie, this may be expected")
-			return
-		}
-		conf := cookieConfig{
-			Name:    cookieAuthorizationName,
-			Value:   signupCookie,
-			Domain:  c.Request.Host,
-			Expires: key.ExpiresAt,
-		}
-		setCookie(c, conf)
-		deleteCookie(c, cookieSignupName, baseDomain)
+func exchangeSignupCookieForSession(c *gin.Context, opts Options) {
+	signupCookie, err := getCookie(c.Request, cookieSignupName)
+	if err != nil {
+		logging.L.Trace().Err(err).Msg("failed to find signup cookie, this may be expected")
+		return
 	}
+
+	exp := time.Now().UTC().Add(opts.SessionDuration)
```

Comment:
Approximating the auth cookie expiry based on the session duration. This should work fine since this will be exchanged within a few minutes.

Reply:
Was this longer duration for the signup cookie necessary for the fix, or is this change related to something else?

Reply:
Ohh, this is for the new cookie. That makes sense! For some reason I thought this was for the signup cookie, but it's clearly not.